src/erlcloud_json.erl

```diff
@@ -8,6 +13,52 @@
 -type decode_value() :: {atom(), JsonField :: binary(), Type :: decode_value_type()}.
 -type decode_value_r() :: {pos_integer(), JsonField :: binary(), Type :: atom()}.
 
+-spec decode_bin(binary()) -> jsx:json_term().
+-spec decode_bin_with_maps(binary()) -> jiffy:json_value().
+-spec encode(jsx:json_term()) -> binary().
+
+-ifdef(ENABLE_NIF_DECODERS).
+
+decode_bin(Binary) ->
+    jiffy_to_jsx(jiffy:decode(Binary, [copy_strings])).
+
+decode_bin_with_maps(Binary) ->
+    jiffy:decode(Binary, [return_maps, copy_strings]).
+
+encode(Term) ->
+    iolist_to_binary(jiffy:encode(jsx_to_jiffy(Term))).
+
+jiffy_to_jsx(List) when is_list(List) ->
+    [ jiffy_to_jsx(X) || X <- List ];
+jiffy_to_jsx({[]}) ->
+    [{}];
+jiffy_to_jsx({List}) ->
+    [ {K, jiffy_to_jsx(V)} || {K, V} <- List ];
+jiffy_to_jsx(V)
+  when is_binary(V); is_number(V); is_boolean(V); V =:= null ->
+    V.
+
+jsx_to_jiffy([{}]) -> {[]};
+jsx_to_jiffy([T|_] = Obj) when is_tuple(T) ->
+    {[ {K, jsx_to_jiffy(V)} || {K, V} <- Obj ]};
+jsx_to_jiffy(List) when is_list(List) ->
+    [ jsx_to_jiffy(V) || V <- List ];
+jsx_to_jiffy(V) ->
+    V.
+
+-else.
+
+decode_bin(Binary) ->
+    jsx:decode(Binary).
```

Comment:
For compatibility with `jsx` 3.0 (and libraries that consume it, "higher up") this should actually be `[{return_maps, false}]`, otherwise (since 3.0 broke interface explicitly) this will be a real issue for consumers.